documentation/modules/ROOT/pages/connectors/postgresql.adoc

```diff
@@ -2262,6 +2263,108 @@ Again, regularly emitting events solves the problem.
+See the following procedure for guidance about how to perform a PostgreSQL database upgrade to enable {prodname} to continue to capture events, and to minimize the risk of data loss.
+
+.Procedure
+
+1. Ensure that no data changes can occur during the upgrade process by temporarily stopping applications that write to the database, or putting them into a read-only mode.
+
+2. Back up the database.
+
+3. Temporarily disable write access to the database.
+
+4. Verify that any changes that occurred in the database before you blocked write operations are saved to the WAL on the replication slot.
+
+5. Provide the connector with enough time to capture all event records that are written to the replication slot. +
+This step ensures that all change events that occured before the downtime are accounted for, and that they are saved to Kafka.
+
+6. Verify that the connector has finished consuming entries from the replication slot by checking the value of the flushed LSN.
+
+7. Shut down the connector gracefully by stopping Kafka Connect. +
+Kafka Connect stops the connectors, flushes all event records to Kafka, and records the last offset received from each connector. +
++
+[NOTE]
+====
+As an alternative to stopping the entire Kafka Connect cluster, you can stop the connector by deleting it.
+Do not remove the offset topic, because it might be shared by other Kafka connectors.
+Later, after you restore write access to the database and you are ready to restart the connector, you must recreate the connector.
+====
+
+8. As a PostgreSQL administrator, drop the replication slot on the primary database server.
+Do not use the xref:postgresql-property-slot-drop-on-stop[`slot.drop.on.stop`] property to drop the replication slot.
+This property is for testing only
+
+9. Stop the database.
+
+10. Perform the upgrade using an approved PostgreSQL upgrade procedure, such as `pg_upgrade`, or `pg_dump` and `pg_restore`.
+
+11. (Optional) Use a standard Kafka tool to remove the connector offsets from the offset storage topic. +
+For an example of how to remove connector offsets, see https://debezium.io/documentation/faq/#how_to_remove_committed_offsets_for_a_connector[how to remove connector offsets] in the {prodname} community FAQ.
+
+12. Restart the database.
+
+13. As a PostgreSQL administrator, create a {prodname} logical replication slot on the database.
+You must create the slot before enabling writes to the database.
+Otherwise, {prodname} cannot capture the changes, resulting in data loss.
+
+ifdef::product[]
++
+For information about setting up a replication slot, see xref:configuring-a-replication-slot-for-the-debezium-pgoutput-plug-in[].
+endif::product[]
+
+14. Verify that the publication that defines the tables for {prodname} to capture is still present after the upgrade.
+If the publication is not available, connect to the database as a PostgreSQL administrator to create a new publication.
+
```

Comment:
@jpechane Reading up on PostgreSQL I learned that a replication slot is created automatically when you create a publication. So if a user finds that the publication is not preserved during the upgrade and it must be recreated, then we can assume that the replication slot is created. Should I reverse the order of steps 13 and 14?

Reply:
I'm not aware of a publication creating a replication slot automatically.  I do know that a _subscription_ will create a remote replication slot on the observed PostgreSQL instance automatically, but a subscription and a publication are two very different things.  

A subscription in this context is a replica telling a primary "hey, I want you to send me data for tables A, B, and C".  

A publication on the other hand is a filter that is optionally associated with a replication slot (decoderbufs doesn't use these), which controls what data the slot is permitted to emit and how the data is to be emitted.  For example, you may have the REPLICA IDENTITY set to FULL for all tables, but a publication that includes tables A and C only means that changes for table B will be dropped by PostgreSQL and never sent to Debezium.